src/browser/cached_resource_fetcher.ts

```diff
@@ -54,37 +57,42 @@ export function cachedFetch(appUuid: string, fileUrl: string, callback: (error:
 
     const appCacheDir = getAppCacheDir(appUuid);
     const filePath = getFilePath(appCacheDir, fileUrl);
+    let err: Error;
+
+    try {
+        await prepDownloadLocation(appCacheDir, filePath);
+        await download(fileUrl, filePath);
+    } catch (e) {
+        err = e;
+    }
+
+    callback(err, filePath);
```

Comment:
In addition to callbacks, may/should/will this async function return value(s). Maybe it's not needed, just asking...